src/CPUActivityAgent.cpp

```diff
@@ -70,13 +71,34 @@ namespace geopm
 
     void CPUActivityAgent::init_platform_io(void)
     {
-        for (int domain_idx = 0; domain_idx < M_NUM_CORE; ++domain_idx) {
+        int scalability_signal_domain = m_platform_io.signal_domain_type("MSR::CPU_SCALABILITY_RATIO");
+        if (scalability_signal_domain < m_freq_ctl_domain_type) {
+            // Set Freq gov domain
+            m_freq_governor->set_domain_type(scalability_signal_domain);
+
+            // update member vars
+            m_freq_ctl_domain_type = m_freq_governor->frequency_domain_type();
+            m_num_freq_ctl_domain = m_platform_topo.num_domain(m_freq_ctl_domain_type);
+
+#ifdef GEOPM_DEBUG
```

Comment:
It's not problematic for a normal run, this was following the example of other agents https://github.com/geopm/geopm/blob/dev/src/PowerBalancer.cpp#L159 have similar throws in debug error cases to inform developers.  

It could be converted to a stderr without issue.